contracts/Witch.sol

```diff
@@ -69,138 +75,205 @@ contract Witch is AccessControl() {
-    function setIlk(bytes6 ilkId, uint32 duration, uint64 initialOffer, uint96 line, uint24 dust, uint8 dec) external auth {
-        require (initialOffer <= 1e18, "Only at or under 100%");
-        ilks[ilkId] = Ilk({
-            duration: duration,
-            initialOffer: initialOffer
-        });
+    /// @param ilkId Id of asset used for collateral
+    /// @param duration Time that auctions take to go to minimal price
+    /// @param initialOffer Proportion of collateral that is sold at auction start (1e18 = 100%)
+    /// @param line Maximum concurrent auctioned collateral
+    /// @param dust Minimum collateral that must be left when buying, unless buying all
+    /// @param dec Multiplying factor (10**dec) for line and dust
+    function setIlk(
+        bytes6 ilkId,
+        uint32 duration,
+        uint64 initialOffer,
+        uint96 line,
+        uint24 dust,
+        uint8 dec
+    ) external auth {
+        require(initialOffer <= 1e18, "Only at or under 100%");
+        ilks[ilkId] = Ilk({duration: duration, initialOffer: initialOffer});
         limits[ilkId] = Limits({
             line: line,
             dust: dust,
             dec: dec,
-            sum: limits[ilkId].sum      // sum is initialized at zero, and doesn't change when changing any ilk parameters
+            sum: limits[ilkId].sum // sum is initialized at zero, and doesn't change when changing any ilk parameters
         });
         emit IlkSet(ilkId, duration, initialOffer, line, dust, dec);
     }
 
-    /// @dev Put an undercollateralized vault up for liquidation.
-    function auction(bytes12 vaultId)
-        external
-    {
-        require (auctions[vaultId].start == 0, "Vault already under auction");
-        require (cauldron.level(vaultId) < 0, "Not undercollateralized");
+    /// @dev Put an undercollateralized vault up for liquidation
+    /// @param vaultId Id of vault to liquidate
+    function auction(bytes12 vaultId) external {
+        require(auctions[vaultId].start == 0, "Vault already under auction");
+        require(cauldron.level(vaultId) < 0, "Not undercollateralized");
 
         DataTypes.Vault memory vault_ = cauldron.vaults(vaultId);
         DataTypes.Balances memory balances_ = cauldron.balances(vaultId);
         Limits memory limits_ = limits[vault_.ilkId];
         limits_.sum += balances_.ink;
-        require (limits_.sum <= limits_.line * (10 ** limits_.dec), "Collateral limit reached");
+        require(limits_.sum <= limits_.line * (10**limits_.dec), "Collateral limit reached");
 
         limits[vault_.ilkId] = limits_;
-        auctions[vaultId] = Auction({
-            owner: vault_.owner,
-            start: block.timestamp.u32()
-        });
+        auctions[vaultId] = Auction({owner: vault_.owner, start: block.timestamp.u32()});
         cauldron.give(vaultId, address(this));
         emit Auctioned(vaultId, block.timestamp.u32());
     }
 
+    /// @dev Private fn called by buy() and payAll()
```

Comment:
What does the function do, though?

Reply:
[Added natspec comment](https://github.com/yieldprotocol/vault-v2/pull/365/commits/a9913aa9dc6e9e9a2f48c9753844ad94156a0a0e), is that good enough?  Or would you like me to refactor or perhaps inline this logic in the main fn's inside of a {} block?

Reply:
hmm actually this fn may go when I implement your `_buy()` suggestion below.